src/Composer/ComposerWidget.vala

```diff
@@ -296,6 +299,26 @@ public class Mail.ComposerWidget : Gtk.Grid {
         });
     }
 
+    private void on_insert_link_clicked () {
+        var insert_link_dialog = new InsertLinkDialog (web_view.get_selected_text ());
+        insert_link_dialog.insert_link.connect (on_link_inserted);
+        insert_link_dialog.transient_for = (Gtk.Window) get_toplevel ();
+        insert_link_dialog.run ();
+    }
+
+    private void on_link_inserted (string url, string title) {
+        var selected_text = web_view.get_selected_text ();
+        if (selected_text != null && title == selected_text) {
+            web_view.execute_editor_command ("createLink", url);
+        } else {
+            if (title != null && title.length > 0) {
+                web_view.execute_editor_command ("insertHTML", """<a href="%s">%s</a>""".printf (url, title));
+            } else {
+                web_view.execute_editor_command ("insertHTML", """<a href="%s">%s</a>""".printf (url, url));
```

Comment:
```suggestion
                web_view.execute_editor_command ("insertHTML", "<a>%s</a>".printf (url));
```

Reply:
Just a tiny thing

Reply:
@tintou I don't think that does what I'd like it to do. An anchor without a href attribute is a placeholder link. Placeholder links are a bit like doing `<a href="#">whatever</a>` (i.e. they don't go anywhere and are designed to be used in Javascript or whatever). I want this to be a real link that goes places, just with the same text as the URL.